riptide-core/src/main/java/org/zalando/riptide/RestBuilder.java

```diff
@@ -1,17 +1,24 @@
 package org.zalando.riptide;
 
 import com.google.common.collect.ImmutableList;
+import org.springframework.core.task.AsyncListenableTaskExecutor;
+import org.springframework.core.task.TaskRejectedException;
 import org.springframework.http.client.AsyncClientHttpRequestFactory;
 import org.springframework.http.client.SimpleClientHttpRequestFactory;
 import org.springframework.http.converter.HttpMessageConverter;
 import org.springframework.scheduling.concurrent.ConcurrentTaskExecutor;
+import org.springframework.util.concurrent.ListenableFuture;
+import org.springframework.util.concurrent.ListenableFutureTask;
 import org.springframework.web.client.RestTemplate;
 
 import javax.annotation.Nullable;
 import java.net.URI;
 import java.util.ArrayList;
 import java.util.List;
+import java.util.concurrent.Callable;
 import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Future;
+import java.util.concurrent.RejectedExecutionException;
```

Comment:
For 4.0.x I'd have to implement a custom executor. Those are relicts of that test. I'll remove it.